### PR TITLE
Abn duration feat

### PR DESCRIPTION
**Pull request by bnika**
Added a duration feature to the UI, the user can now choose the speed of the generated speech from 0.8-1.5

### CODE REVIEW

**Comment by lumpidu** [comment]
It might be interesting to see for recent jobs, which voice has been used

File: app/services/tts_service.rb [hunk 1]
```diff
@@ -9,18 +9,34 @@ class TtsService < ApplicationService
   # TODO: make configurable
   #@@tts_url = 'http://127.0.0.1:8000/v0/speech'
 
-@@tts_url = 'https://api.grammatek.com/tts/v0/speech'
+  @@tts_url = 'https://api.grammatek.com/tts/v0/speech'
```
Comment (lumpidu) [inline app/services/tts_service.rb:12]: indentation

File: app/services/tts_service.rb [hunk 2]
```diff
@@ -9,18 +9,34 @@ class TtsService < ApplicationService
...
 # a directory where the response audio is written to as .mp3
 @@tmp_audio_path = "./app/assets/audio/"
 
-  def initialize(text, format, filename)
+  def initialize(text, format, filename, duration, voice)
+    if @@voices_dict.key?(voice)
+      voice_id = @@voices_dict[voice]
+    else
+      # Let's not make the whole thing break, but use a default voice
+      voice_id= "Alfur"
```
Comment (lumpidu) [inline app/services/tts_service.rb:33]: nitpick: space before `=`

File: app/services/tts_service.rb [hunk 3]
```diff
@@ -54,14 +70,15 @@ def call
...
+  def init_request_data(text, format, duration, voice)
     # The request data format as required by the currently used TTS-service.
     {"Engine": "standard",
      "LanguageCode": "is-IS",
      "OutputFormat": "mp3",
+     "Duration": "#{duration}",
      "SampleRate": "22050",
-     "TextType": "#{format}",
-     "VoiceId": "Alfur",
+     "TextFormat": "#{format}",
```
Comment (lumpidu) [inline app/services/tts_service.rb:80]: Has this changed inside the API interface ?

File: app/services/tts_service.rb [hunk 4]
```diff
@@ -9,18 +9,34 @@ class TtsService < ApplicationService
...
 
-  def initialize(text, format, filename)
+  def initialize(text, format, filename, duration, voice)
+    if @@voices_dict.key?(voice)
+      voice_id = @@voices_dict[voice]
+    else
+      # Let's not make the whole thing break, but use a default voice
+      voice_id= "Alfur"
+    end
+    p "=================== USING VOICE: #{voice_id} ======================"
```
Comment (lumpidu) [inline app/services/tts_service.rb:35]: Shouldn't this be `Rails.logger.debug` ?

File: app/services/tts_service.rb (hunk 3 above)
Comment (bnika) [reply app/services/tts_service.rb:80]: yes, to prevent a name clash with the ssml-text-type

File: app/services/tts_service.rb (hunk 4 above)
Comment (bnika) [reply app/services/tts_service.rb:35]: probably ...

File: app/services/tts_service.rb (hunk 3 above)
Comment (lumpidu) [reply app/services/tts_service.rb:80]: ok, but then we are not compatible anymore with the original web service and we need to change also the OpenAPI spec

**Comment by lumpidu** [comment]
LGTM

**Pull request merged**